src/pool/pool_disjoint.c

```diff
@@ -135,25 +133,32 @@ static void destroy_slab(slab_t *slab) {
         LOG_ERR("deallocation of slab data failed!");
     }
 
-    umf_ba_global_free(slab->chunks);
     umf_ba_global_free(slab);
 }
 
-// return the index of the first available chunk, SIZE_MAX otherwise
 static size_t slab_find_first_available_chunk_idx(const slab_t *slab) {
-    // use the first free chunk index as a hint for the search
-    for (bool *chunk = slab->chunks + slab->first_free_chunk_idx;
-         chunk != slab->chunks + slab->num_chunks_total; chunk++) {
-
-        // false means not used
-        if (*chunk == false) {
-            size_t idx = chunk - slab->chunks;
-            LOG_DEBUG("idx: %zu", idx);
-            return idx;
+    // Calculate the number of 64-bit words needed.
+    size_t num_words = (slab->num_chunks_total + 63) / 64;
+    for (size_t i = 0; i < num_words; i++) {
+        // Invert the word: free bits (0 in the allocated mask) become 1.
+        uint64_t word = ~(slab->chunks[i]);
+
+        // For the final word, clear out bits that exceed num_chunks_total.
+        if (i == num_words - 1) {
+            size_t bits_in_last_word = slab->num_chunks_total - (i * 64);
+            if (bits_in_last_word < 64) {
+                word &= (((uint64_t)1 << bits_in_last_word) - 1);
+            }
+        }
+        if (word != 0) {
+            size_t bit_index = utils_get_rightmost_set_bit_pos(word);
+            size_t free_chunk = i * 64 + bit_index;
```

Comment:
define `64` as a bitmap value size of something

Reply:
done